doc/sphinx/proofs/writing-proofs/equality.rst

```diff
@@ -39,10 +39,19 @@ Tactics for simple equalities
 
 .. tacn:: reflexivity
 
-   For a goal with the form :n:`{? forall @open_binders , } t = u`,
-   verifies that `t` and `u` are
-   :term:`definitionally equal <definitional equality>`, and if so,
-   solves the goal (by applying `eq_refl`).  If not, it fails.
+   After doing an :tacn:`intros` and :ref:`βδi-reducing <conversion-rules>` constant expressions,
+   if the resulting goal is an equality n:`t = u` in which `t` and `u` are identical, the tactic
+   proves the goal (by applying `eq_refl`).  If not, it fails.
+
+   For example, `n * n = n * n` or `1 = 1 + 2` or `1 + 2 + n = 3 + n` or `0 + n = n` all work.
+   (`n + 1 + 2 = n + 3` doesn't work because the left hand side is represented as `(n + 1) + 2`,
+   which won't be reduced.)
+
+   TODO: doesn't handle `n + 0 = n' or `1 * n = n` - how can I explain this?
+
+   TODO: Why simpl of `1 * n = n` is `n + 0 = n` while simpl of `n * 1 = n` makes no change?
```

Comment:
Same reason as above, this time multiplication is defined in terms of addition and it is the first argument that is being reduced at each "computation" step. When that first argument is a closed term like `1`, Coq is able to simplify it, but when it is a variable in the context like `n`, it is not able to reduce it. Hence why in this case, induction would be required.